test/test_ops.py

```diff
@@ -38,6 +43,89 @@ def test_unsupported_dtypes(self, device, dtype, op):
             for sample in samples:
                 op(sample.input, *sample.args, **sample.kwargs)
 
+    # Helper for comparing torch tensors and numpy arrays
+    def assertEqualHelper(self, actual, expected, msg, *, dtype, exact_dtype=True, **kwargs):
+        assert isinstance(actual, torch.Tensor)
+
+        # Some NumPy functions return scalars, not arrays
+        if isinstance(expected, Number):
+            self.assertEqual(actual.item(), expected, **kwargs)
+        elif isinstance(expected, np.ndarray):
+            # Handles exact dtype comparisons between arrays and tensors
+            if exact_dtype:
+                # Allows array dtype to be float32 when comparing with bfloat16 tensors
+                # since Numpy doesn't support the bfloat16 dtype
+                # Also ops like scipy.special.erf, scipy.special.erfc etc. promote float16
+                # to float32
+                if expected.dtype == np.float32:
+                    assert actual.dtype in (torch.float16, torch.bfloat16, torch.float32)
+                else:
+                    assert expected.dtype == torch_to_numpy_dtype_dict[actual.dtype]
+
+            self.assertEqual(actual,
+                             torch.from_numpy(expected).to(actual.dtype),
+                             msg,
+                             exact_device=False,
+                             **kwargs)
+        else:
+            self.assertEqual(actual, expected, msg, exact_device=False, **kwargs)
+
+    def _test_reference_numerics(self, dtype, op, tensors, equal_nan=True):
+        def _helper_reference_numerics(expected, actual, msg, exact_dtype, equal_nan=True):
+            if not torch.can_cast(numpy_to_torch_dtype_dict[expected.dtype.type], dtype):
+                exact_dtype = False
+
+            if dtype in [torch.uint8, torch.int8, torch.bool]:
+                self.assertEqualHelper(actual, expected, msg, dtype=dtype,
+                                       exact_dtype=exact_dtype, rtol=1e-3, atol=1e-2)
+
+            elif dtype is torch.bfloat16:
```

Comment:
Since we know we're comparing with NumPy I think skipping bfloat16 by default is an option, if it makes things simpler